tests/instrument/expr_match.ml.reference

```diff
@@ -99,3 +99,174 @@ let f x =
         (match x with
          | Foo  -> (___bisect_mark___expr_match 36; print_endline "foobar")
          | Bar  -> (___bisect_mark___expr_match 37; print_endline "barbar"))))
+let f x =
+  ___bisect_mark___expr_match 43;
+  (match x with
+   | Foo |Bar  as ___bisect_matched_value___ ->
+       ((((match ___bisect_matched_value___ with
+           | Foo  -> (___bisect_mark___expr_match 41; ())
+           | Bar  -> (___bisect_mark___expr_match 42; ())
+           | _ -> ()))[@ocaml.warning "-8-11"]);
+        print_endline "foo"))
```

Comment:
I think there are actually some other problems, e.g. let's say `s : unit -> bool` is a side-effecting function and you have

```ocaml
match v with
| Foo | Bar when s () -> ...
| _ -> ...
```

If you rewrite this to e.g.

```ocaml
match v with
| Foo when s () -> ...
| Bar when s () -> ...
| _ -> ...
```

You will cause the side effect twice. It's getting late so I am not sure if there is something else you could obviously rewrite this to, that wouldn't have such a problem, but I don't think so right now.